Userland/Libraries/LibTest/TestCase.h

```diff
@@ -27,10 +47,71 @@ class TestCase : public RefCounted<TestCase> {
     {
     }
 
+    TestCase(DeprecatedString const& name, TestFunction& fn, bool is_benchmark)
+        : m_name(name)
+        , m_function(move(fn))
```

Comment:
Does moving from references work? This doesn't seem like it would work.

Reply:
I was mostly copypasting from the existing code above (the other constructor); I admittedly have very little understanding of this aspect of C++.

I believe I was forced to add this constructor because of what we're returning in `randomized()` below. Advice needed!

Reply:
It seems like you just need to change `randomized()`'s `return make_ref_counted<TestCase>(name, test_case_function, false);` to `return make_ref_counted<TestCase>(name, move(test_case_function), false);`.

I'm also pretty sure that it would be a user-after-free otherwise, since you are handing off a reference to a stack-allocated value.